app/src/main/java/com/example/campusguide/MapsActivity.kt

```diff
@@ -40,33 +41,6 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback {
         mapFragment.getMapAsync(this)
         ObjectBox.init(this.applicationContext)
 
-        val currentLocationButton: FloatingActionButton = findViewById(R.id.currentLocationButton)
-        currentLocationButton.setOnClickListener {
-            //Check if location permission has been granted
-            if(ActivityCompat.checkSelfPermission(this, Manifest.permission.ACCESS_FINE_LOCATION) == PackageManager.PERMISSION_GRANTED) {
-                goToCurrentLocation()
-            } else {
-                //Request location permission
-                ActivityCompat.requestPermissions(this, arrayOf(Manifest.permission.ACCESS_FINE_LOCATION), LOCATION_PERMISSION_ACCESS_CODE)
-            }
-        }
-
-        val navigateButton = findViewById<FloatingActionButton>(R.id.navigateButton)
-        navigateButton.setOnClickListener {
-            val getDirectionsDialogFragment =
-                GetDirectionsDialogFragment(
-                    GetDirectionsDialogFragment.DirectionsDialogOptions(
-                        null, null,
-                        EmptyDirectionsGuard(this,
-                            CallbackDirectionsConfirmListener { start, end ->
-                                //Display the directions time
-                                route.set(start, end)
-                            })
-                    )
-                )
-            getDirectionsDialogFragment.show(supportFragmentManager, "directionsDialog")
-        }
-
```

Comment:
Code was eaten by rebase. Should be re-added

Reply:
this code got moved to the bottom of the MapsActivity file into 2 functions: setCurrentLocationListener() and setNavButtonListener(). it's still in the file, just somewhere else! setButtonListeners() makes sure to call all this code so it's not getting ignored either :)

also putting it in the onCreate makes that function wayyyy too long and messy

Reply:
Okay! Then it's all good. Good to know